homeassistant/components/fritz/switch.py

```diff
@@ -561,16 +561,9 @@ def __init__(self, fritzbox_tools: FritzBoxTools, device: FritzDevice) -> None:
         """Init Fritz profile."""
         super().__init__(fritzbox_tools, device)
         self._attr_is_on: bool = False
-
-    @property
-    def unique_id(self) -> str:
-        """Return device unique id."""
-        return f"{self._mac}_switch"
-
-    @property
-    def icon(self) -> str:
-        """Return device icon."""
-        return "mdi:router-wireless-settings"
+        self._name = f"{device.hostname} Internet Access"
+        self._attr_unique_id = f"{self._mac}_internet_access"
+        self._attr_icon = "mdi:router-wireless-settings"
```

Comment:
How about minimising this PR and just changing icon and unique id in their respective property?

```suggestion

     @property
     def unique_id(self) -> str:
         """Return device unique id."""
         return f"{self._mac}_internet_access"

      @property
      def icon(self) -> str:
          """Return device icon."""
          return "mdi:router-wireless-settings"
```

Reply:
The current suggested solution by @chemelli74 is preferred.
Your suggestion should not be applied.

Reply:
Ok thanks for the feedback @frenck!

Reply:
The icon is statically and doesn't change, thus can be moved into the class attribute:

```py
class FritzBoxProfileSwitch(FritzDeviceBase, SwitchEntity):
    """Defines a FRITZ!Box Tools DeviceProfile switch."""
    
    _attr_icon = "mdi:router-wireless-settings"
```

```suggestion
        self._attr_unique_id = f"{self._mac}_internet_access"
```

Reply:
Done